index.bs

```diff
@@ -958,6 +959,8 @@ The <dfn>device type</dfn> indicates the kind of device used for the context. It
 <dd>Provides the broadest compatibility and usability across all client devices with varying degrees of performance.</dd>
 <dt>"<dfn enum-value>gpu</dfn>"</dt>
 <dd>Provides the broadest range of achievable performance across graphics hardware platforms from consumer devices to professional workstations.</dd>
+<dt>"<dfn enum-value>npu</dfn>"</dt>
+<dd>Provides narrower functionality on dedicated machine learning devices which perform between CPU and GPU, favoring other factors over raw performance like lower power usage for longer battery life or lower latency for results. The backend implementation may fall back to other devices for certain operators and parts of the graph.</dd>
```

Comment:
> The backend implementation may fall back to other devices for certain operators and parts of the graph.

That this may be true of the "gpu" device type, too, as @philloooo pointed out in https://github.com/webmachinelearning/webnn/issues/623#issuecomment-2025767956

Reply:
True - added same comment to "gpu".